Fix knx tests broken by Python 3.14.3 asyncio changes

### Description

The following `knx` tests are flaky/failing due to the Python 3.14.3 asyncio executor scheduling change (see home-assistant/core#162263 for background).

CPython PR python/cpython#142358 changed how `run_in_executor` delivers results back to the event loop. Results are now set directly on the asyncio future instead of going through `call_soon_threadsafe`, which means `await` returns immediately but other `call_soon_threadsafe` callbacks may not have run yet.

#### Affected tests

**`tests/components/knx/test_light.py::test_light_rgb_individual` (failed 1/10)**

```
tests/components/knx/test_light.py:539: in test_light_rgb_individual
    await knx.assert_read(test_red_state)
tests/components/knx/conftest.py:231: in assert_read
    await self.assert_telegram(group_address, None, GroupValueRead, ignore_order)
tests/components/knx/conftest.py:188: in assert_telegram
    raise AssertionError(
E   AssertionError: No Telegram found. Expected: GroupValueRead - 1/1/4 - None
```

**`tests/components/knx/test_sensor.py::test_sensor_restore` (failed 4/10)**

```
tests/components/knx/test_sensor.py:91: in test_sensor_restore
    knx.assert_state("sensor.test", RESTORED_STATE, **RESTORED_STATE_ATTRIBUTES)
tests/components/knx/conftest.py:76: in assert_state
    assert test_state.state == state
E   AssertionError: assert 'unknown' == '21.0'
E     
E     - 21.0
E     + unknown
```


#### What was done

`test_sensor_restore` has been temporarily disabled with `@pytest.mark.xfail` in PR #169079.

Note: only `test_sensor_restore` was marked as xfail. `test_light_rgb_individual` was not disabled because many other light tests follow the same `assert_read`-after-setup pattern and could be similarly affected — it likely indicates a broader timing issue in the KNX test fixtures rather than a problem isolated to this one test. Since it only occured in 1/10 runs, I left it as is. Please still look at the test if possible and see if you can reproduce it and fix it.

#### What needs to happen

The codeowner should investigate and fix the underlying timing issue in these tests. Common fixes include:
- Adding `await hass.async_block_till_done()` after service calls
- Converting sync entity methods to async
- Restructuring mocks to avoid unintended executor code paths

### Additional context

- Related PR: home-assistant/core#162263
- Disable PR: #169079

--- a/tests/components/knx/conftest.py
+++ b/tests/components/knx/conftest.py
@@ -97,6 +97,7 @@ class KNXTestKit:
                 state=XknxConnectionState.CONNECTED,
                 connection_type=XknxConnectionType.TUNNEL_TCP,
             )
+            await self.hass.async_block_till_done()
 
         def knx_ip_interface_mock():
             """Create a xknx knx ip interface mock."""

--- a/tests/components/knx/test_sensor.py
+++ b/tests/components/knx/test_sensor.py
@@ -62,9 +62,6 @@ async def test_sensor(hass: HomeAssistant, knx: KNXTestKit) -> None:
     await knx.assert_no_telegram()
 
 
-@pytest.mark.xfail(
-    reason="Flaky due to Python 3.14.3 asyncio changes - see home-assistant/core#162263"
-)
 async def test_sensor_restore(hass: HomeAssistant, knx: KNXTestKit) -> None:
     """Test restoring KNX sensor state."""
     ADDRESS = "2/2/2"

